0003-lightning-as-alpha-ledger.adoc

```diff
@@ -0,0 +1,279 @@
+| `β::Identity`
+| The identity on beta that **B** will be transferred to when the beta-HTLC is activated with the correct secret
+
+| `secret_hash`
+| `hex-encoded-bytes`
+| The output of calling `hash_function` on the secret
+
+|===
+
+[start=2]
+2. Bob _accepts_ the request and performs the following step:
+    .. Create a hold invoice using `addholdinvoice` with the amount `X` and the hashed preimage `secret_hash` (https://github.com/lightningnetwork/lnd/blob/aa1cd04dbf07a9195d5ada752f383988d8d01fa7/cmd/lncli/invoicesrpc_active.go#L142[more details]).
+    .. Bob _subscribes_ to the invoice and waits for the payment either using
+      ... `SubscribeSingleInvoice` - this is not available through the CLI but as RPC; or
+      ... `LookupInvoice` - this is available trough the CLI but needs to be polled regularly.
+3. Bob response to Alice's request with the following information:
+
+[cols=3*,options=header]
+|===
+| Name
+| JSON Encoding
+| Description
+
+| `alpha_redeem_identity`
+| `α::Identity`
+| The identity on alpha that **A**, this is the LN `invoice`
+
+| `beta_refund_identity`
+| `β::Identity`
+| The identity on beta that **B** will be transferred to when the beta-HTLC is activated after `beta_expiry`
+|===
+
+[start=4]
+4. Alice now starts the https://github.com/comit-network/RFCs/blob/master/RFC-003-SWAP-Basic.md#1-alice-deploys-%CE%B1-htlc[execution phase] by paying the invoice using the LND command `sendpayment`.
+5. Bob gets notified about funding of alpha (i.e. the invoice has been paid but cannot be settled yet), and continues with https://github.com/comit-network/RFCs/blob/master/RFC-003-SWAP-Basic.md#2-bob-deploys-%CE%B2-htlc[deploying beta-HTLC], i.e. he deploys a HTLC on Ethereum.
+6. As soon as beta has enough confirmations for Alice, she redeems the beta-HTLC using her secret.
+7. Bob gets notified about this, learns the secret and can now settle the LND invoice by invoking the LND command `settleInvoice` (https://github.com/lightningnetwork/lnd/blob/aa1cd04dbf07a9195d5ada752f383988d8d01fa7/cmd/lncli/invoicesrpc_active.go#L53[more details]).
+
+The trade is complete.
+
+== Spike Outcome
+
+=== Lightning Network: a new ledger
+Similiar to the ledger definitions for https://github.com/comit-network/RFCs/blob/master/RFC-004-Bitcoin.md[Bitcoin] and https://github.com/comit-network/RFCs/blob/master/RFC-006-Ethereum.md[Ethereum] we need to handle the Lightning Network differently.
+This is required because the comit-node and btsieve need to perform different actions accordingly.
+We are always talking about Ledgers and Assets, (e.g. _Bitcoin_ Asset on the _Bitcoin_ Ledger, _Ether_ Asset on the _Ethereum_ Ledger, _Erc20_ on the _Ethereum_ Ledger, ...), Hence,
+if we follow this approach, for supporting LN (through LND) we will need to introduce a new pair of Ledger and Asset:
+
+* Ledger: the **Lightning Network**. _Ledgers_ are used as _settlement layers_ for our HTLCs. In the case of LND, this layer is the Lightning Network.
```

Comment:
:x: I think this spike should include, how the ledger is exactly represented.

With the LN, we have node IDs and channels. A channel is a construct between two nodes.
It looks like the `Ledger` definition cannot be a node ID, it has to be the channel id, otherwise which node ID do you use? Alice's or Bob's?
If we refer to the channel, Alice aswell as Bob should be able to query their local node to see if they are in the specified channel and can also find out the other party's node ID (at least I hope LND exposes this information :upside_down_face:)

This in turn makes the alpha redeem and success identity redundant, since both would just either the channel or node ID which are already known information at this point.

Reply:
I see no added value in considering channels ids in the protocol and implementation.
I do not see why Alice and Bob should query their node for a specific channel.
Started to play channel ids just add unjustified restrictions on what route should a payment come from.

I agree, identity should be node id.

Bolt Lightning over Bitcoin Core is the ledger.

Reply:
> It looks like the Ledger definition cannot be a node ID, it has to be the channel id, otherwise which node ID do you use? Alice's or Bob's?

What do you mean with _which node ID do you use_ or _`Ledger` definition cannot be a node ID, it has to be the channel id_? 

> Bolt Lightning over Bitcoin Core is the ledger.

I agree on this. The channel does not seem useful to me, i.e. Alice has no guarantees to see all channels of Bob nor can she/or should she send a payment to a particular channel. Payments on LN are identified by the node IDs. 

Think differently, if Bob provides Alice a channel ID - a channel between Bob and Claire. What use is that for Alice? All she can do is verify that this channel exist, but nothing about its state, nor can she send a payment to that channel.

However, if we limit ourselves to 0-hop channels (Alice <-> Bob) than the channel might be useful, but that's not what we want.

Reply:
@D4nte 

> Bolt Lightning over Bitcoin Core is the ledger.

With or without any identifiers of the nodes/channels?
We talked about this topic quite a lot already and explored different options. To me, this spike doesn't yet clarify, which of those options we should now go for and what the implications are.

Reply:
@bonomat 

> Think differently, if Bob provides Alice a channel ID - a channel between Bob and Claire. What use is that for Alice? All she can do is verify that this channel exist, but nothing about its state, nor can she send a payment to that channel.
However, if we limit ourselves to 0-hop channels (Alice <-> Bob) than the channel might be useful, but that's not what we want.

True, I was only thinking 0-hop channels.

Reply:
Without any identifier to nodes or channels.

There are no identifier to Bitcoin Core Ledger except for the name `bitcoin` and the network (e.g. `mainnet`).

I expect the Lightning BOLT Bitcoin Core ledger to have the same amount of information to be identified: a name (e.g. `LightningBitcoin`) and a network (e.g. `mainnnet`).

Reply:
Thanks! That is the information I was looking for :)